Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -101,14 +111,16 @@ def self.audit_args
   def self.audit
     args = audit_args.parse
 
+    new_cask = args.new? || args.new_cask?
+    new_formula = args.new? || args.new_formula?
```

Comment:
```suggestion
    new_cask = args.new?
    if args.new_cask?
      # odeprecated "brew audit --new-cask"
      new_cask = true
    end
    new_formula = args.new?
    if args.new_formula?
      # odeprecated "brew audit --new-formula"
      new_formula = true
    end
```

Reply:
@reitermarkus Still think this would be good to do so we delete and print deprecation messages on usage.

Reply:
`odeprecated` is automatically done by `CLI::Parser` once the `replacement`s are uncommented on the `switch` calls.

Reply:
Yup but we're reliant on the commenting and changing of `odeprecated` to `odisabled` process to get this code removed.

Reply:
I already added a `odeprecated` comment above on the `switch` calls. Once the `switch` calls use `disable: true`, calling `args.new_formula?`/`args.new_cask?` here will fail CI anyways.

Reply:
Also added a comment and `disable: false` to make it greppable.